asistente_ladm_col/lib/geometry.py

```diff
@@ -907,39 +908,113 @@ def fix_boundaries(self, layer, id_field):
+        # Select buildings overlaps with plots
+        processing.run("native:selectbylocation", {'INPUT': input_layer,
+                                                   'PREDICATE': [5],  # overlaps
+                                                   'INTERSECT': intersect_layer,
+                                                   'METHOD': 0})
+        input_overlaps = input_layer.selectedFeatureIds()
+
+        # Select buildings to are within plots
+        processing.run("native:selectbylocation", {'INPUT': input_layer,
+                                                   'PREDICATE': [6],  # are within
+                                                   'INTERSECT': intersect_layer,
+                                                   'METHOD': 0})
+        input_within = input_layer.selectedFeatureIds()
+
+        return input_disjoint, input_overlaps, input_within
+
+    @staticmethod
+    def get_intersection_features(layer, geometries, id_field=None):
+        """
+        return a list of list, every list has the feature ids of features
+        that intersect with every geometry.
+        If id_field is None return id of QGIS
+        layer: QgsVectorLayer
+        geometries: list of QgsGeometries (Order is not modify)
+        """
+        if id_field:
+            id_field_idx = layer.fields().indexFromName(id_field)
+            request = QgsFeatureRequest().setSubsetOfAttributes([id_field_idx])
+            dict_features = {feature.id(): feature for feature in layer.getFeatures(request)}
+        else:
+            dict_features = {feature.id(): feature for feature in layer.getFeatures()}
 
-        return buildings_with_no_plot, buildings_not_within_a_single_plot
+        index = QgsSpatialIndex(layer)
+
+        intersecting_ids = list()
+        for geometry in geometries:
+            bbox = geometry.boundingBox()
+            bbox.scale(1.001)
+            candidates_ids = index.intersects(bbox)
+            candidate_features = [dict_features[candidate_id] for candidate_id in candidates_ids]
+            feature_ids = list()
+            for candidate_feature in candidate_features:
+                candidate_geometry = candidate_feature.geometry()
+                if geometry.intersects(candidate_geometry):
+                    if id_field:
+                        feature_ids.append(candidate_feature[id_field])
+                    else:
+                        feature_ids.append(candidate_feature.id())
+            intersecting_ids.append(feature_ids)
+        return intersecting_ids
+
+    @staticmethod
+    def get_polygon_nodes_layer(polygon_layer, id_field):
+        """
+        Layer is created with unique vertices. It is necessary because 'remove duplicate vertices' processing
+        algorithm does not filter the data as we need them
+        """
+        tmp_plot_nodes_layer = processing.run("native:extractvertices", {'INPUT': polygon_layer, 'OUTPUT': 'memory:'})['OUTPUT']
```

Comment:
Couldn't we call remove duplicate vertices or delete duplicate geometries?

Reply:
Yes, It is better to use the qgis algorithms

Reply:
Thanks, resolve in 91dfcc8